lib/omniauth/strategies/custom_identity.rb

```diff
@@ -69,6 +69,8 @@ def callback_phase
         return fail_with_log!(:too_many_login_attempts) if too_many_login_attempts?
 
         if identity.present?
+          # Give signup_state a chance to record the link if one exists
+          signup_state.on_user_authentication(identity.user) if signup_state
```

Comment:
Can you talk to me about why this is here?  I didn't expect to see this here.

Reply:
This accommodates the flow where a user arrives with trusted data and then signs in.   After they sign in we need to link the account to the provided uuid so we can auto-redirect them back in subsequent visits.

I believe `custom_identity` is the only code that touches the user after they authenticate and before they're redirected back.  Maybe this should go into a routine though given your comment above about not fattening the `SignupState` model.

Reply:
I am fairly sure that a user who logs in will pass back through `SessionsCreate`, specifically the [handle_during_login](https://github.com/openstax/accounts/blob/master/app/handlers/sessions_create.rb#L62-L76) method.  This handler also has access to the `signup_state` (passed in from the controller), so it is probably a good place to add the `UserExternalUuid` if it doesn't exist yet.  Also, I believe that the `CustomIdentity` stuff modified here is only used for password logins, so social logins may miss out on the current way of linking external UUIDs (whereas `SessionsCreate` should not).  

It would be nice to have a spec that shows that someone logging in socially gets their external UUID added.  Unfortunately looking around, we may not have any specs that test successful social login, tho we do have some unsuccessful ones.  Search `with_omniauth_test_mode`.

Reply:
Yep, you're correct!  the code there isn't needed since `SessionsCreate` calls the `TransferSignupState`  routine